Lib/fontTools/ttLib/tables/TupleVariation.py

```diff
@@ -446,6 +444,118 @@ def getTupleSize_(flags, axisCount):
 			size += axisCount * 4
 		return size
 
+	def getCoordWidth(self):
+		""" Return 2 if coordinates are (x, y) as in gvar, 1 if single values
+		as in cvar, or 0 if empty.
+		"""
+		firstDelta = next((c for c in self.coordinates if c is not None), None)
+		if firstDelta is None:
+			return 0  # empty or has no impact
+		if type(firstDelta) is tuple and len(firstDelta) == 2:
+			return 2
+		elif type(firstDelta) in (int, float):
+			return 1
+		else:
+			raise TypeError("invalid type of delta: %s" % type(firstDelta))
+
+	def scaleDeltas(self, scalar):
+		if scalar == 1.0:
+			return  # no change
+		coordWidth = self.getCoordWidth()
+		if coordWidth == 2:
+			self.coordinates = [
+				(d[0] * scalar, d[1] * scalar) if d is not None else None
+				for d in self.coordinates
+			]
+		elif coordWidth == 1:
+			self.coordinates = [
+				d * scalar if d is not None else None
+				for d in self.coordinates
+			]
+
+	def roundDeltas(self):
+		coordWidth = self.getCoordWidth()
```

Comment:
Should coordWidth be computed just once and then assigned to field?

Reply:
the thing is, the way it's defined, this coordWidth can actually change as result of `optimize` (it can become 0 if all deltas are set to None), or `__iadd__` (if self coordWidth is 0 and other has coordWidth 1) or `calcInferredDeltas` (coordWidth can change from 0 to 2). So we need to compute it each time.